ratpack-exec/src/main/java/ratpack/exec/internal/DefaultExecController.java

```diff
@@ -210,6 +221,18 @@ private DefaultExecution createExecution(Action<? super Execution> initialExecut
           throw new InternalRatpackError("could not start execution", e);
         }
       }
+
+      private CharSequence getLabel(DefaultExecution.Ref parentRef) {
+        // always call these as we want to increment the counters regardless of whether
+        // an explicit label has been set
+        int forkedExecutionNumber;
+        if (parentRef != null) {
+          forkedExecutionNumber = parentRef.execution.nextChild();
+        } else {
+          forkedExecutionNumber = topLevelExecutionCounter.getAndIncrement();
+        }
+        return maybeLabel.orElse(String.valueOf(forkedExecutionNumber));
```

Comment:
so, if I fork 2 executions and each of those forks a child, then the resulting 2 executions will have the same label since we are not combing the parent label with the child's.
Is this the desired behavior? Should we be doing something like `<parent>-<child>` instead?

Reply:
If you had nested forking couldn't you still do this with a <parent>-<child> model? It is entirely possible with (clumsy design most likely) to yield something vaguely like this:

![image](https://user-images.githubusercontent.com/391345/63220622-61de9680-c151-11e9-8d81-5f2e9118a904.png)